### PR TITLE
feat: add syntax for equality constraints

### PR DESCRIPTION
**Pull request by mlutze**
related to #5647

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala [hunk 1]
```diff
@@ -281,6 +281,14 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       optional(keyword("with") ~ WS ~ oneOrMore(TypeConstraint).separatedBy(optWS ~ "," ~ optWS)) ~> ((o: Option[Seq[ParsedAst.TypeConstraint]]) => o.getOrElse(Seq.empty))
     }
 
+    def EqualityConstraint: Rule1[ParsedAst.EqualityConstraint] = rule {
+      SP ~ Type ~ optWS ~ "~" ~ optWS ~ Type ~ SP ~> ParsedAst.EqualityConstraint
+    }
+
+    def OptEqualityConstraintList: Rule1[Seq[ParsedAst.EqualityConstraint]] = rule {
+      optional(keyword("where") ~ WS ~ oneOrMore(EqualityConstraint).separatedBy(optWS ~ "," ~ optWS)) ~> ((o: Option[Seq[ParsedAst.EqualityConstraint]]) => o.getOrElse(Seq.empty))
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Parser.scala:289]: Hmm.

But, in any case, such eq constraints are likely to be exceedingly rare?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:289]: Perhaps this would be sufficient:

```
with Eq[a], a ~ b
```

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:289]: But let us worry about it later.

Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:289]: Yes this syntax is not meant to be definitive, just enough to get off the ground.

**Pull request merged**